app/domain/payment_provider.rb

```diff
@@ -42,6 +42,16 @@ def HPB
     check_bank_public_keys!(bank_x, bank_e)
   end
 
+  def XTC(document)
+    raise ArgumentError.new('document is empty') if document.nil? || document.empty?
+
+    client.send(:upload, PaymentProviders::Xtc, document)
+  end
+
+  def Z54(from = nil, to = nil)
```

Comment:
ich wäre für `since` und `until`, weil das eher auf Zeiträume hinweist. `from` und `to` hört sich so nach Orten an.